packetbeat/flows/worker_test.go

```diff
@@ -127,3 +127,114 @@ func TestCreateEvent(t *testing.T) {
 		}
 	}
 }
+
+func Test_shouldKillFlow(t *testing.T) {
+	logp.TestingSetup()
+
+	// Build biflow event.
+	vlan := uint16(171)
+	mac1 := []byte{1, 2, 3, 4, 5, 6}
+	mac2 := []byte{6, 5, 4, 3, 2, 1}
+	ip1 := []byte{203, 0, 113, 3}
+	ip2 := []byte{198, 51, 100, 2}
+	port1 := uint16(38901)
+	port2 := uint16(80)
+
+	id := newFlowID()
+	id.AddEth(mac1, mac2)
+	id.AddVLan(vlan)
+	id.AddIPv4(ip1, ip2)
+	id.AddTCP(port1, port2)
+	processor := &flowsProcessor{}
+
+	type args struct {
+		flow                 *biFlow
+		flowProcessorTimeout time.Duration
+		currentTime          time.Time
+		activeTimeoutEnabled bool
+	}
+	tests := []struct {
```

Comment:
perhaps a test case where active timeout is enabled, but the idle time out hits.
I assume this can happen when a flow is started but no traffic is seen.
i.e. flow.ts == flow.createTS
🤷